docs/getting-started.md

```diff
@@ -0,0 +1,517 @@
+the `Task` definition is constant and the value of parameters can change during runtime.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: build-and-push-docker-image-from-git
+spec:
+  params:
+    - name: BUILDER_IMAGE
+      description: The location of the builder image
+      default: quay.io/buildah/stable:v1.14.3
+  resources:
+    inputs:
+      - name: source
+        type: git
+    outputs:
+      - name: image
+        type: image
+  steps:
+    - name: build-and-push
+      image: $(inputs.params.BUILDER_IMAGE)
+      workingDir: /workspace/source
+      command: ["/bin/bash"]
+      args:
+        - -c
+        - |
+          set -e
+          IMAGE_NAME="$(outputs.resources.image.url)"
+          buildah bud --tls-verify="false" --layers -f "./Dockerfile" -t "$IMAGE_NAME:latest" .
+          buildah push --tls-verify="false" "$IMAGE_NAME:latest" "docker://$IMAGE_NAME:latest"
+      securityContext:
+        privileged: true
+      volumeMounts:
+        - name: varlibcontainers
+          mountPath: /var/lib/containers
+  volumes:
+    - name: varlibcontainers
+      emptyDir: {}
+```
+
+### Configuring `Task` execution credentials
+
+Before you can execute your `TaskRun`, you must create a `secret` to push your image
+to your desired image registry:
+
+**Note:** You can get your Docker access token at [Docker Hub](https://hub.docker.com/settings/security).
+
+```bash
+kubectl create secret docker-registry regcred \
+                    --docker-server=docker.io \
+                    --docker-username=<your-name> \
+                    --docker-password=<your-token> \
+                    --docker-email=<your-email>
+```
+
+You must specify a `ServiceAccount` that uses this `secret` to execute your `TaskRun`:
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: tutorial-service
+secrets:
+  - name: regcred
+```
+
+Save the `ServiceAccount` definition above to a file and apply the YAML file to make the `ServiceAccount` available for your `TaskRun`:
+
+```bash
+kubectl apply -f <name-of-file.yaml>
+```
+
+### Running your `Task`
+
+You are now ready for your first `TaskRun`!
+
+A `TaskRun` binds the inputs and outputs to already defined `PipelineResources`, sets values
+for variable substitution parameters, and executes the `Steps` in the `Task`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: build-and-push-docker-image-from-git-task-run
+spec:
+  serviceAccountName: tutorial-service
+  taskRef:
+    name: build-and-push-docker-image-from-git
+  resources:
+    inputs:
+      - name: source
+        resourceRef:
+          name: hello-world-git
+    outputs:
+      - name: image
+        resourceRef:
+          name: hello-world-image
+```
+
+Save the YAML files that contain your Task, and `PipelineResource` definitions and apply them using the following command:
+
+```bash
+kubectl apply -f <name-of-file.yaml>
```

Comment:
Again, suggest showing each file applied individually to demonstrate correct order of operations.